document-generator-api/src/test/java/uk/gov/companieshouse/document/generator/api/document/render/RenderDocumentRequestHandlerTest.java

```diff
@@ -69,41 +83,87 @@ public void setUp() throws IOException{
 
         requestParameters = new HashMap<>();
         requestParameters.put("resource_uri", RESOURCE_URI);
+        requestParameters.put("resource_id", RESOURCE_ID);
         requestParameters.put("request_id", REQUEST_ID);
-
-        when(mockHttpConnectionHandler.openConnection(any(String.class))).thenReturn(mockHttpURLConnection);
     }
 
     @Test
     @DisplayName("Send the data to the render service successfully")
-    public void testSendDataToRenderServiceSuccess() throws IOException {
+    public void testSendDataToRenderServiceSuccess() throws IOException, RenderServiceException {
 
+        setValidOpenConnection();
         when(convertJsonHandler.convert(any(String.class))).thenReturn("long data");
 
-        setMockHttpConnectionForSuccess(201);
-
+        setMockHttpConnectionForSuccess(HTTP_CREATED);
         RenderDocumentResponse response = renderDocumentRequestHandler.sendDataToDocumentRenderService(
                 "http://www.test.com", renderDocumentRequest, requestParameters);
 
         assertEquals(PDF_LOCATION, response.getLocation());
 
         verifyHttpConnectionMock(true);
-        assertEquals(201, response.getStatus());
+        assertEquals(HTTP_CREATED, response.getStatus());
     }
 
     @Test
     @DisplayName("Send the data to the render service and obtain a response error")
-    public void testSendDataToRenderServiceServerResponseError() throws IOException {
+    public void testSendDataToRenderServiceServerResponseError() throws IOException, RenderServiceException {
 
-        setMockHttpConnectionForError(500);
+        setValidOpenConnection();
+
+        setMockHttpConnectionForError(HTTP_INTERNAL_ERROR);
         RenderDocumentResponse response = renderDocumentRequestHandler.sendDataToDocumentRenderService(
                 "http://www.test.com", renderDocumentRequest, requestParameters);
 
         assertNull(response.getDocumentSize());
         assertNull(response.getLocation());
 
         verifyHttpConnectionMock(false);
-        assertEquals(500, response.getStatus());
+        assertEquals(HTTP_INTERNAL_ERROR, response.getStatus());
+    }
+
+    @Test
+    @DisplayName("Error Thrown when opening connection fails")
+    public void testErrorThrownWhenOpeningConnectionFails() throws IOException {
+
+        when(mockHttpConnectionHandler.openConnection(any(String.class))).thenThrow(IOException.class);
+
+        assertThrows(RenderServiceException.class, () -> renderDocumentRequestHandler.sendDataToDocumentRenderService(
+                "http://www.test.com", renderDocumentRequest, requestParameters));
```

Comment:
We shouldn't use a link for a real website in our unit tests, can we remove this reference `http://www.test.com`, also this should be a Constant as its used several times.